src/ray/gcs/gcs_client/service_based_gcs_client.cc

```diff
@@ -191,6 +193,21 @@ void ServiceBasedGcsClient::ReconnectGcsServer() {
   int index = 0;
   for (; index < RayConfig::instance().ping_gcs_rpc_server_max_retries(); ++index) {
     if (get_server_address_func_(&address)) {
+      // After GCS is restarted, the gcs client will reestablish the connection. At
+      // present, every failed RPC request will trigger `ReconnectGcsServer`. In order to
+      // avoid repeated connections in a short period of time, we add a protection
+      // mechanism: if the address does not change, the connection can be made at most
+      // once in `minimum_gcs_reconnect_interval_milliseconds` milliseconds.
+      if (last_reconnect_address_ == address &&
+          (current_sys_time_ms() - last_reconnect_timestamp_ms_) <
+              RayConfig::instance().minimum_gcs_reconnect_interval_milliseconds()) {
+        RAY_LOG(DEBUG)
```

Comment:
Tiny but important: If reconnect request keeps under amount of thousands, we can change log level to INFO/WARNING to indicate that we have jumped over the reconnection.

Reply:
SGTM